llvm/lib/Target/AMDGPU/SIPeepholeSDWA.cpp

```diff
@@ -391,21 +490,23 @@ MachineInstr *SDWASrcOperand::potentialToConvert(const SIInstrInfo *TII,
     if (!Reg->isReg() || !Reg->isDef())
       return nullptr;
 
-    for (MachineInstr &UseMI : getMRI()->use_nodbg_instructions(Reg->getReg()))
-      // Check that all instructions that use Reg can be converted
-      if (!isConvertibleToSDWA(UseMI, ST, TII) ||
-          !canCombineSelections(UseMI, TII))
+    // Check that all instructions that use Reg can be converted
+    for (MachineInstr &UseMI :
+         getMRI()->use_nodbg_instructions(Reg->getReg())) {
+      MachineInstr *SrcMI = findUseSrc(UseMI, *Reg, getMRI());
```

Comment:
This is looking at the uses of Reg->getReg() inside a loop over the uses of Reg->getReg()?

Reply:
No, see also my other [comment](https://github.com/llvm/llvm-project/pull/133087/files/243157eb5f2437c6c8e6f9cecbbdc173a2e8c363#r2023045944); `regSequenceFindSingleSubregUse` does not iterate the uses of `Reg->getReg()` again (which is `%6` in the example from the other comment). It iterates the uses of the register defined by `UseMI` (`%7` in the example) if it is a `REQ_SEQUENCE`.